### ISSUE TITLE
Issues with 3rd party compilation order

### ISSUE THREAD

**Issue by lars18th** [canonical/multipass#3802]
**Describe the bug**
Manually compiling Multipass the build process stops at some point.

**To Reproduce**
Inside one Ubuntu 20.04 container, the build process compiling with CLang stops with this error:
```
-- Looking for openssl/des.h - not found
CMake Error at 3rd-party/libssh/libssh/ConfigureChecks.cmake:80 (message):
  Could not detect openssl/des.h
Call Stack (most recent call first):
  3rd-party/libssh/CMakeLists.txt:51 (include)
```

**Expected behavior**
The problem seems to be related to the _order_ of the 3rd party components:
https://github.com/canonical/multipass/blob/6c4b8b4f23462a58c81b708808782e682627bf9b/CMakeLists.txt#L125-L138

If you check the pre-builded `vcpkg-ports` 3rd party package, it includes custom `grpc` and `poco` packages as well. However, the configuration is not using these libs to compile other 3rd party packages. And the same is true for the `libssh`.

The idea is then: compile first custom `libssh`, `grpc` and `poco`; and after use _only_ these versions to compile all the other dependencies.

I hope you want to improve the compilation process to reduce dependencies and use _only_ the custom build 3rd party libraries. This could facilitate compilation in other environments and reduce compilation time.

**Comment by giuliazanchi** [comment]
Hi @lars18th , thanks for bringing this to our attention, it's definitely worth exploring further to simplify and improve the build process and, possibly, the build instructions. 
Since you seem to have investigated the problem and possible solutions, feel free to open a PR and suggest a fix, we're always happy to review and incorporate contributions from the community.

**Comment by ljonathan0719** [comment]
Hello! Is there any way I could contribute to this issue?

**Comment by ricab** [comment]
Hi @ljonathan0719. Yes, we welcome community contributions, so you can submit a pull request! I suggest starting by finding a simple recipe to reproduce the issue and take it from there (if you haven't yet). Thank you for your interest and let us know if you need further guidance.

**Comment by ricab** [comment]
@ljonathan0719 do you mean to pursue #3939 any further? If not, I guess this would be free @SuhasHareesh.

**Comment by zizeca** [comment]
I think you need to use `add_dependencies(...)` to build in a specific order
[add_dependencies](https://cmake.org/cmake/help/latest/command/add_dependencies.html)

**Comment by vicdeveloperr** [comment]
I don't see movement here for a while.
I'm working on this.

**Comment by vicdeveloperr** [comment]
Why this issue is relevant? I don't have this problem compiling in the latest ubuntu LTS. And this is a no longer supported version.

**Comment by lars18th** [comment]
> Why this issue is relevant? 

Because we want to compile in _any_ Linux (compatible) environment, not only in supported versions.

Please, try to think on this:

- _Why do we need 3party libraries if we compile for a _specific_ Linux distribution?_
 In this case, the correct strategy would be to install the 3party library on the system and compile with it.

**Comment by vicdeveloperr** [comment]
I understand now. Thanks @lars18th

**Comment by vicdeveloperr** [comment]
I have the following error, trying to generate the build system in Ubuntu 20.04:
```
CMake Error at 3rd-party/vcpkg/scripts/buildsystems/vcpkg.cmake:893 (_find_package):
  By not providing "FindQt6.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt6", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt6" with any of
  the following names:

    Qt6Config.cmake
    qt6-config.cmake

  Add the installation prefix of "Qt6" to CMAKE_PREFIX_PATH or set "Qt6_DIR"
  to a directory containing one of the above files.  If "Qt6" provides a
  separate development package or SDK, be sure it has been installed.
Call Stack (most recent call first):
  CMakeLists.txt:182 (find_package)
```
I'm not sure if it's related to this issue.

**Comment by tobe2098** [comment]
No, this is related to cmake not being able to find your qt6 installation. How did you install qt and  where is that installation? It is possible to tell cmake about the Qt6 installation with a flag.

Edit: it is `-DCMAKE_PREFIX_PATH=`

**Comment by rawmeat21** [comment]
Is this issue still open? I'd like to work on this.

**Comment by tobe2098** [comment]
Hi @rawmeat21, feel free to start working on this, but I would recommend you find a recipe to reproduce this issue first and post it here. If you cannot reproduce the issue it may be that we should close the issue itself.

**Comment by rawmeat21** [comment]
Hello @tobe2098 , I wanted to ask if this issue is still relevant, I noticed a comment by @vicdeveloperr that they don't see the issue on the latest LTS version of Ubuntu. I can try looking into this issue, but is it worth fixing? I'm kinda new to contributing to open source so I wanted to ask.

**Comment by tobe2098** [comment]
If you experience the issue or reproduce it, then yes. Our build system has changed a lot in recent times, so I suspect the issue will not be present anymore, but feel free to confirm it. Helping close this issue is already a great help :)
